h/tweens.py

```diff
@@ -72,3 +72,19 @@ def csrf_tween(request):
         return response
 
     return csrf_tween
+
+
+def content_security_policy_tween_factory(handler, registry):
+    policy = registry.settings.get('csp', {})
+    policy = "; ".join([
+        " ".join([k] + [v2 for v2 in v if v2 is not None])
+        for k, v in sorted(policy.items())
+        if [v2 for v2 in v if v2 is not None]
+    ])
+
+    def content_security_policy_tween(request):
+        resp = handler(request)
+        resp.headers["Content-Security-Policy-Report-Only"] = policy.format(request=request)
```

Comment:
I'd suggest adding a `csp.report_only` setting that can be set with a header -- `CSP_REPORT_ONLY`, say, and then toggling the name of the header (`Content-Security-Policy` vs `Content-Security-Policy-Report-Only`) on the basis of that setting.